src/content/blog/astro-4140.mdx

```diff
@@ -0,0 +1,271 @@
+
+<BlogContentImage src={frontmatterIntellisense} />
+
+To enable content file intellisense, add the following to your Astro config:
+
+```js
+import { defineConfig } from 'astro';
+
+export default defineConfig({
+	experimental: {
+		contentIntellisense: true
+	}
+})
+```
+
+and in VS Code, enable the `astro.content-intellisense` setting. For other editors, pass `contentIntellisense: true` to the initialization parameters of the Astro language server.
+
+```json
+{
+	"astro.content-intellisense": true
+}
+```
+
+This feature currently only works in the editor, and will be enabled in `astro check` in a future release.
+
+## Deprecate support for dynamic `prerender` values
+
+In order to improve our bundling and tree-shaking capabilities, Astro 4.14 deprecates support for using dynamic values for the `prerender` option in your Astro pages.
+
+This feature can now be replaced with the new `astro:route:setup` hook in Astro integrations, which allows you to dynamically control the prerendering of specific pages.
+
+```diff
+---
+- export const prerender = import.meta.env.PROD;
+---
+```
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  integrations: [setPrerender()],
+});
+
+function setPrerender() {
+  return {
+    name: 'set-prerender',
+    hooks: {
+      'astro:route:setup': ({ route }) => {
+        if (route.component.endsWith('/blog/[slug].astro')) {
+          route.prerender = true;
+        }
+      },
+    },
+  };
+}
+```
+
+## New `injectTypes` integration API
+
+Type generation has become all the rage in recent years, and Astro is no exception! We use type generation pretty extensively for features like Astro DB and Content Collections (or the new Content Layer API).
+
+Thanks to [@florian-lefebvre](https://github.com/florian-lefebvre), Astro integrations can now join the fun with the new `injectTypes` integration API! This API allows integrations to inject types into the user's project, making it easier to provide type definitions for your integration's features.
+
+```js
+// my-integration/index.js
+export default {
+	name: 'my-integration',
+	'astro:config:done': ({ injectTypes }) => {
+		injectTypes({
+			filename: "types.d.ts",
+			content: "declare module 'virtual:my-integration' {}"
+		})
+	}
+};
+```
+
+For more information on this feature, check out the [Astro integration API documentation](https://docs.astro.build/en/reference/integrations-reference/#injecttypes-options).
+
+## Support for metastrings in the Code component
+
+Astro 4.14 adds support for the `meta` attribute in the `Code` component. This attribute simulate using properties on code blocks in Markdown, for example `` ```js astro=cool ``, which can then be used by [Shiki transformers](https://shiki.matsu.io/guide/transformers#meta) to apply custom transformations to the code block.
```

Comment:
wording nit
```suggestion
Astro 4.14 adds support for the `meta` attribute in the `Code` component. This attribute simulates using properties on code blocks in Markdown, for example `` ```js astro=cool ``, which can then be used by [Shiki transformers](https://shiki.matsu.io/guide/transformers#meta) to apply custom transformations to the code block.
```